src/osd/ECBackend.cc

```diff
@@ -2364,34 +2386,35 @@ struct CallClientContexts :
     ceph_assert(res.returned.size() == to_read.size());
     ceph_assert(res.errors.empty());
     for (auto &&read: to_read) {
-      pair<uint64_t, uint64_t> adjusted =
-	ec->sinfo.offset_len_to_stripe_bounds(
-	  make_pair(read.get<0>(), read.get<1>()));
-      ceph_assert(res.returned.front().get<0>() == adjusted.first);
-      ceph_assert(res.returned.front().get<1>() == adjusted.second);
+      auto bounds = make_pair(read.get<0>(), read.get<1>());
+      auto aligned = ec->sinfo.offset_len_to_stripe_bounds(bounds);
+      if (aligned.first != read.get<0>() || aligned.second != read.get<1>()) {
+        aligned = ec->sinfo.offset_len_to_chunk_bounds(bounds);
```

Comment:
Why not align to chunk_bounds in the first place?  If offset_len_to_stripe_bounds returned the same offset and len, offset_len_to_chunk_bounds should as well as anything that is stripe aligned is also chunk aligned.

Reply:
Hmm, I'm actually not sure what's going on here.  to_read should be the original user offset/length, right?  res is the chunks read from the shards.  Seems like with this implementation we should be either aligning to stripe or to chunk depending on whether the read was submitted as a partial_read.  I'm not sure I understand how this conditional addresses that.